src/util/cmdline.cpp

```diff
@@ -145,6 +148,42 @@ const char *cmdlinet::getval(const char *option) const
   return value->second.front().c_str();
 }
 
+std::string cmdlinet::expand_path_string(std::string path) const
```

Comment:
why do you need this function?

Reply:
This function is a placeholder (now changed to a better name: `expand_user`) that will expand the `~` into the home directory of the user. This effectively gets replaced by `$HOME` on UNIX and `%userprofile%` on Windows.